keras_nlp/models/deberta/deberta_encoder.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DeBERTa encoder block implementation based on `keras.layers.Layer`."""
+
+from tensorflow import keras
+
+from keras_nlp.models.deberta.disentangled_self_attention import (
+    DisentangledSelfAttention,
+)
+from keras_nlp.utils.keras_utils import clone_initializer
+
+from keras_nlp.layers.transformer_layer_utils import (  # isort:skip
+    merge_padding_and_attention_mask,
+)
+
+
+class DebertaEncoder(keras.layers.Layer):
```

Comment:
Should we call this `DisentangledAttentionEncoder` in keeping with the other layer name? `DebertaEncoder` feels a big confusing as you could easily call the whole backbone the "encoder"